### PR TITLE
Fix go to defintion for Sbt's Scala sources

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/metals/issues/1981

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/Compilers.scala [hunk 1]
```diff
@@ -488,9 +488,12 @@ class Compilers(
...
     val modifiedInput =
       originInput.copy(value = withOuter)
-    val adjustLspData = AdjustLspData.create(pos => {
-      new LspPosition(pos.getLine() - 1, pos.getCharacter() - ident.size)
-    })
+    val adjustLspData = AdjustLspData.create(
+      pos => {
+        new LspPosition(pos.getLine() - 1, pos.getCharacter() - ident.size)
+      },
+      filterOutLocations = { loc => loc.getUri().isWorksheet }
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/Compilers.scala:495]: we shouldn't adjust positions outside of the worksheets.

File: metals/src/main/scala/scala/meta/internal/metals/AdjustLspData.scala [hunk 2]
```diff
@@ -77,6 +72,24 @@ trait AdjustLspData {
   }
 }
 
+case class AdjustLspPos(
```
Comment (ckipp01) [inline metals/src/main/scala/scala/meta/internal/metals/AdjustLspData.scala:75]: I normally don't love case classes that are verbs. What do you think of `AdjustedLspPos`?

File: metals/src/main/scala/scala/meta/internal/metals/Compilers.scala (hunk 1 above)
Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/metals/Compilers.scala:495]: 🤔 not work for now, but I'm wondering for clarity in the future if we should try to unify where we have adjustments done. For example we check here for worksheets and sbt and adjust accordingly. However, Ammonite also has some adjustments that we do, but those live in 	`Ammonite.scala`. I can see how that could be confusing from someone on the outside. Or, is it a good distinction to have that adjustments related to the PC are in `AdjustLspData` and other things live in their corresponding places? What do you think?

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/Compilers.scala:495]: Yeah, I will move the adjustments to separate locations

File: metals/src/main/scala/scala/meta/internal/metals/AdjustLspData.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/AdjustLspData.scala]: Fixed!

**Pull request merged**